### PR TITLE
Added support - NFC tags

### PR DESCRIPTION
**Pull request by ashwinpilgaonkar**
Task/Issue URL: https://github.com/duckduckgo/Android/issues/851
Tech Design URL: N/A
CC: 

**Description**:
On scanning an NFC tag that holds web URL, the DuckDuckGo Android app was unable able to open the URL. This PR now fixes that issue. I needed to set up a new `<intent-filter>` in `Manifest.xml`, that would look for `android.nfc.action.NDEF_DISCOVERED`.

**Steps to test this PR**:
1. Ideally should be tested using an NFC tag that holds a web URL.

2. I tested my changes by building another small application that sends a `android.nfc.action.NDEF_DISCOVERED` intent, thus mocking a physical NFC tag. I used the following code snippet to achieve this - 
```
                Intent intent = new Intent();
                intent.setAction("android.nfc.action.NDEF_DISCOVERED");
                intent.setData(Uri.parse("https://www.example.com"));
                startActivity(intent);
```
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by JeffAment** [comment]
I happen to have a spare NFC tag, so I tested this for you; everything seems to work as intended. Good work! :smile:

**Comment by cmonfortep** [comment]
Thanks @JeffAment !!!! You appeared at the right moment here, magical!

btw, how was the user experience? I guess the first time you scan a tag, a system dialog appears requesting nfc permission, right? or does it work automatically without any user prompt?

**Comment by ashwinpilgaonkar** [comment]
> btw, how was the user experience? I guess the first time you scan a tag, a system dialog appears requesting nfc permission, right? or does it work automatically without any user prompt?
>
I think the NFC permission is granted directly at the time of installation. It isn't requested during runtime like other permissions such as location, storage, etc.

When I tested it, it automatically works without any user prompt. Once the intent is sent, the app opens the URL. I'm assuming it works the same way with an actual NFC tag. @JeffAment would probably be able to give us an accurate answer to this.

**Comment by JeffAment** [comment]
> Thanks @JeffAment !!!! You appeared at the right moment here, magical!

No problem, @cmonfortep; happy to help! Hocus-pocus :tophat: :rabbit: :tada:

> When I tested it, it automatically works without any user prompt. Once the intent is sent, the app opens the URL. I'm assuming it works the same way with an actual NFC tag. @JeffAment would probably be able to give us an accurate answer to this.

I can confirm that there was no permission prompt for me. I also checked the "App permissions" screen after successfully using the NFC feature, and it doesn't mention NFC permissions anywhere, as do other browsers with NFC support, so I believe that @ashwinpilgaonkar is correct in saying that the permission is granted directly at install-time without user intervention.

I would also like to mention that my Google Pixel XL Android 10 successfully opens URLs stored on the NFC tag even when the `android.permission.NFC` permission is omitted from the manifest. The official documentation on this is vague, but I'm thinking that the permission might not be necessary for this instance since it's just a simple intent filter for which Android does the heavy lifting. I'm not necessarily advocating for removing the permission, but I just wanted to note this. FWIW if it were up to me, I'd probably leave it in to avoid potential headaches down the road.

**Comment by ashwinpilgaonkar** [comment]
> I would also like to mention that my Google Pixel XL Android 10 successfully opens URLs stored on the NFC tag even when the `android.permission.NFC` permission is omitted from the manifest. The official documentation on this is vague, but I'm thinking that the permission might not be necessary for this instance since it's just a simple intent filter for which Android does the heavy lifting.
>
I'm not a 100% sure about this, but I think that the NFC permission is essential for other NFC related activities - such as switching on WiFi/Bluetooth when you tap an NFC device. Probably isn't required for opening URLs though.

> I'm not necessarily advocating for removing the permission, but I just wanted to note this. FWIW if it were up to me, I'd probably leave it in to avoid potential headaches down the road.
>
Yes, I completely agree with you on that one.

**Comment by cmonfortep** [comment]
Thanks both, @JeffAment and @ashwinpilgaonkar, since our goal here is just to open URLs from NFC tags, I want to double-check if that permission is needed or not. We will merge this PR anyways (with or without permission), agree with Jeff that keeping the permission will remove problems in the future, but before that, I just want to double-check if we need it for our current usage.

Two main reasons:
1. AFAIK, new permission in the manifest that are accepted during installation will block auto-updates for users, till they go to GPlay and accept that permission.
2. Optics are important in our app. We don't want to request something that we don't really need, that could also confuse the user when they see that request from us.

I shared the question internally, will report back here once we know something else.

Thanks both again!

**Comment by cmonfortep** [comment]
Hi there again! I managed to get some NFC tags and do some deep testing with a few devices to validate if we need that NFC permission. It seems we don't need it (🎉 ) in order to accept NFC intents, so I would remove that permission from the manifest and we are good to go.

sounds good @ashwinpilgaonkar ?

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.NFC"/>
```
Comment (cmonfortep) [inline app/src/main/AndroidManifest.xml:13]:
```suggestion
```

**Review by cmonfortep — CHANGES_REQUESTED** [review]
We can safely remove this permission from the manifest.

**Comment by ashwinpilgaonkar** [comment]
> Hi there again! I managed to get some NFC tags and do some deep testing with a few devices to validate if we need that NFC permission. It seems we don't need it (🎉 ) in order to accept NFC intents, so I would remove that permission from the manifest and we are good to go.
> 
> sounds good @ashwinpilgaonkar ?

Yep, I've made the required changes.

**Review by cmonfortep — APPROVED** [review]
LGTM Thanks!

**Pull request merged**